src/averageAges.js

```diff
@@ -55,7 +69,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const callBackSon = (sum, child) =>
+    sum + (child.born
+      - people.find(mother => mother.name === child.mother).born);
+
+  const callBackFilter = (child) =>
+    onlyWithSon === true
+      ? people.some(mother => child.mother === mother.name
+          && child.sex === 'm')
+      : people.some(mother => child.mother === mother.name);
```

Comment:
also, move this to the separate function & reuse it
```suggestion
  const callBackFilter = (child) => onlyWithSon
      ? hasMother(people, child.mother) && child.sex === 'm'
      : hasMother(people, child.mother);
```